curves/bls12_377/src/lib.rs

```diff
@@ -56,3 +56,15 @@ const SCALE_VALIDATE: Validate = Validate::Yes;
 pub const SCALE_USAGE: Usage = ark_scale::make_usage(SCALE_COMPRESS, SCALE_VALIDATE);
 
 type ArkScale<T> = ark_scale::ArkScale<T, SCALE_USAGE>;
+
+/// Curve hooks available for user implementation.
+///
+/// All parameters and return values are SCALE encoded.
+pub trait CurveHooks: 'static {
+    fn bls12_377_multi_miller_loop(a: Vec<u8>, b: Vec<u8>) -> Result<Vec<u8>, ()>;
+    fn bls12_377_final_exponentiation(f12: Vec<u8>) -> Result<Vec<u8>, ()>;
+    fn bls12_377_msm_g1(bases: Vec<u8>, scalars: Vec<u8>) -> Result<Vec<u8>, ()>;
+    fn bls12_377_mul_projective_g1(base: Vec<u8>, scalar: Vec<u8>) -> Result<Vec<u8>, ()>;
+    fn bls12_377_msm_g2(bases: Vec<u8>, scalars: Vec<u8>) -> Result<Vec<u8>, ()>;
+    fn bls12_377_mul_projective_g2(base: Vec<u8>, scalar: Vec<u8>) -> Result<Vec<u8>, ()>;
+}
```

Comment:
It's fine to move this of course, but just fyi..

We originally kept some other files in subdirectories like src/curve so that arkworks changes could more easily be reflected here.  I doubt that works anymore, but maybe.. 

It's not relevant here of course, but mentioning it since it's the one thing you're moving to lib.rs